grid-cli/cmd/deploy_kubernetes.go

```diff
@@ -186,7 +186,7 @@ var deployKubernetesCmd = &cobra.Command{
 			masterNode = uint32(nodes[0].NodeID)
 		}
 		master.Node = masterNode
-		if workersNode == 0 && len(workers) > 0 {
+		if len(workersNode) < len(workers) {
 
 			filter, disks, rootfss := filters.BuildK8sFilter(
 				workers[0],
```

Comment:
line 203 there's no need to filter with len(workers), if there's some nodes in workers nodes, then you should filter for the workers that were not assigned any nodes